### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by victorgz**
## Description

- Upgrade metadata API version
- Add new eslint plugin for Lightning Web Security (validation for #106)
- - **NOTE:** eslint LWS rules are disabled in some tests. Validation is not required as they are not deployed to an org
- Update gitignore to exclude SFDX Local Development Server files

## How Has This Been Tested?

Local tests + scratch org

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: force-app/main/default/lwc/.eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": ["@salesforce/eslint-config-lwc/recommended"]
+  "extends": [
+    "@salesforce/eslint-config-lwc/recommended",
+    "@locker/eslint-config-locker"
```
Comment (scolladon) [inline force-app/main/default/lwc/.eslintrc.json:4]: **question:** why using the basic one ?

And why not using the [recommended one](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.security_lwsec_tools_lint#:~:text=If%20you%20want%20notification) ?

File: force-app/sample/default/lwc/.eslintrc.json [hunk 2]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": ["@salesforce/eslint-config-lwc/recommended"]
+  "extends": [
+    "@salesforce/eslint-config-lwc/recommended",
+    "@locker/eslint-config-locker"
```
Comment (scolladon) [inline force-app/sample/default/lwc/.eslintrc.json:4]: **question:** same question here

**Review by scolladon — COMMENTED** [review]
I have one question!

**Comment by victorgz** [comment]
@scolladon fyi : I've also updated the PMD version (and some depecrated rules with the new version) + fixed the scratch-org-def issue we talked about in the other PR 😄

File: force-app/main/default/lwc/.eslintrc.json (hunk 1 above)
Comment (scolladon) [reply force-app/main/default/lwc/.eslintrc.json:5]: Maybe we could also add the recommended one for locker ?

```suggestion
    "@locker/eslint-config-locker/recommended"
```

Comment (victorgz) [reply force-app/main/default/lwc/.eslintrc.json:5]: Ah yes! in fact I missunderstood your first question. New commit is coming, thanks mate !

**Review by scolladon — APPROVED** [review]
LGTM

**Pull request merged**